frontend/src/metabase/visualizations/components/ObjectDetail/ObjectDetailsTable.tsx

```diff
@@ -115,8 +140,23 @@ export function DetailsTable({
   onVisualizationClick,
   visualizationIsClickable,
 }: DetailsTableProps): JSX.Element {
-  const { cols } = data;
-  const row = zoomedRow;
+  const { cols, row } = React.useMemo(() => {
+    const { cols } = data;
+    const columnSettings = settings["detail.columns"];
+    const columnIndexes = columnSettings
+      .filter((columnSetting: any) => columnSetting.enabled)
+      .map((columnSetting: any) =>
+        findColumnIndexForColumnSetting(cols, columnSetting),
+      )
+      .filter(
+        (columnIndex: number) => columnIndex >= 0 && columnIndex < cols.length,
+      );
+
+    return {
+      cols: columnIndexes.map((i: number) => cols[i]) as any[],
+      row: columnIndexes.map((i: number) => zoomedRow[i]),
+    };
+  }, [data.cols, zoomedRow, settings]);
```

Comment:
Eslint complaints we need the whole `data` in the dependencies array. Can we do something to get rid of the warning, but without disabling our dep array check completely?